canonical/avalanche-k8s-operator
# add release automation

### This pull request:

- Splits Push and PR actions into their dedicated GHA workflows.
- Adds automation to push to edge on merge to main.
- Makes git checkouts shallow to shave off some execution time.

### Testing Instructions
- Make sure it passes CI. 
- Make sure the charm was released properly after merge.
